docs/source/library-user-guide/table-constraints.md

```diff
@@ -0,0 +1,46 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Table Constraint Enforcement
+
+Table providers can describe table constraints using the
+[`TableConstraint`] and [`Constraints`] APIs. These constraints include
+primary keys, unique keys, foreign keys and check constraints.
+
+DataFusion does **not** currently enforce these constraints at runtime.
+They are provided for informational purposes and can be used by custom
+`TableProvider` implementations or other parts of the system.
+
+- **Nullability**: The only property enforced by DataFusion is the
+  nullability of each [`Field`] in a schema. Columns marked as not
+  nullable should not produce null values during execution. DataFusion
+  does not check this when data is ingested.
```

Comment:
It might help to mention here what happens if a column marked as non nullable return null values during execution. Specifically, I think it generates a runtime error. Something like this perhaps

```suggestion
- **Nullability**: The only property enforced by DataFusion is the
  nullability of each [`Field`] in a schema. Returning data with null values
  for Columns marked as not nullable will result in runtime errors  during execution. DataFusion
  does not check or enforce nullability when data is ingested.
```